### PR TITLE
Split backends into their own projects

### PR DESCRIPTION
**Pull request by Beyley**

### CODE REVIEW

File: Furball.Vixie.Backends.OpenGL.Shared/BufferObjectGL.cs [hunk 1]
```diff
@@ -182,7 +183,7 @@ public static unsafe BufferObjectGL CreateNew<pDataType>(IGLBasedBackend backend
         /// Unlocks the Buffer, so that other buffers can be bound
         /// </summary>
         /// <returns>Self, used for chaining Methods</returns>
-        internal BufferObjectGL Unlock() {
+        public BufferObjectGL Unlock() {
```
Comment (Eeveelution) [inline Furball.Vixie.Backends.OpenGL.Shared/BufferObjectGL.cs:186]: Locking is intentionally internal only, this used to/is used to prevent a Shader/Buffer/Whatever switches during which they shouldn't ever be switched

File: Furball.Vixie.Backends.OpenGL.Shared/ShaderGL.cs [hunk 2]
```diff
@@ -149,7 +150,7 @@ public class ShaderGL : IDisposable {
         /// Unlocks the Shader, so that other Shaders can be bound
         /// </summary>
         /// <returns>Self, used for chaining Methods</returns>
-        internal ShaderGL Unlock() {
+        public ShaderGL Unlock() {
```
Comment (Eeveelution) [inline Furball.Vixie.Backends.OpenGL.Shared/ShaderGL.cs:153]: same here with the internal lock

File: Furball.Vixie.Backends.OpenGL.Shared/TextureGL.cs [hunk 3]
```diff
@@ -79,7 +81,7 @@ public class TextureGL : Texture, IDisposable {
         /// <summary>
         /// Used for determening whether or not to Flip it internally because it's a framebuffer
         /// </summary>
-        internal bool IsFramebufferTexture;
+        public bool IsFramebufferTexture;
```
Comment (Eeveelution) [inline Furball.Vixie.Backends.OpenGL.Shared/TextureGL.cs]: this should defnitly not be opened up to the end user to change however they like, internal fits better here.

File: Furball.Vixie.Backends.OpenGL.Shared/VertexArrayObjectGL.cs [hunk 4]
```diff
@@ -106,7 +107,7 @@ public class VertexArrayObjectGL : IDisposable {
         /// Unlocks the Texture, so that other Textures can be bound
         /// </summary>
         /// <returns>Self, used for chaining Methods</returns>
-        internal VertexArrayObjectGL Unlock() {
+        public VertexArrayObjectGL Unlock() {
```
Comment (Eeveelution) [inline Furball.Vixie.Backends.OpenGL.Shared/VertexArrayObjectGL.cs:110]: same here

File: Furball.Vixie.Backends.Shared/TextureRenderTarget.cs [hunk 5]
```diff
@@ -1,14 +1,9 @@
 using System.Numerics;
-using Furball.Vixie.Graphics.Backends;
 
-namespace Furball.Vixie.Graphics {
+namespace Furball.Vixie.Backends.Shared {
     public abstract class TextureRenderTarget {
         public abstract Vector2 Size { get; protected set; }
 
-        public static TextureRenderTarget Create(uint width, uint height) {
```
Comment (Eeveelution) [inline Furball.Vixie.Backends.Shared/TextureRenderTarget.cs:8]: why was this removed?

File: Furball.Vixie.Helpers/DisposeQueue.cs [hunk 6]
```diff
@@ -21,7 +21,7 @@ public static class DisposeQueue {
             }
         }
 
-        internal static void DisposeAll() {
+        public static void DisposeAll() {
```
Comment (Eeveelution) [inline Furball.Vixie.Helpers/DisposeQueue.cs:24]: DisposeQueue should never have been made public.

File: Furball.Vixie/Texture.cs [hunk 7]
```diff
@@ -1,14 +1,8 @@
...
-
-namespace Furball.Vixie.Graphics {
-    public abstract class Texture {
-        public abstract Vector2 Size { get; protected set; }
-        public int Width => (int)Size.X;
-        public int Height => (int)Size.Y;
+using Furball.Vixie.Backends.Shared;
 
+namespace Furball.Vixie {
+    public class Textures {
```
Comment (Eeveelution) [inline Furball.Vixie/Texture.cs]: why was this renamed to textures????

File: Furball.Vixie/TextureRenderTargets.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+using Furball.Vixie.Backends.Shared;
+
+namespace Furball.Vixie {
+    public class TextureRenderTargets {
```
Comment (Eeveelution) [inline Furball.Vixie/TextureRenderTargets.cs]: same with this why was this renamed to TextureRenderTargets???

**Review by Eeveelution — CHANGES_REQUESTED** [review]
I don't agree on some things getting opened up to the end user and I seriously don't understand why shit was moved from Texture to Textures and TextureRenderTarget to TextureRenderTargets, it was in the exact same class as before to be used as a way to call new Texture which since it's an abstract class can't be done anymore so instead it calls into the backend, and I see absolutely no reason for the change

**Comment by Beyley** [comment]
> I don't agree on some things getting opened up to the end user and I seriously don't understand why shit was moved from Texture to Textures and TextureRenderTarget to TextureRenderTargets, it was in the exact same class as before to be used as a way to call new Texture which since it's an abstract class can't be done anymore so instead it calls into the backend, and I see absolutely no reason for the change

Splitting without changing the class name is not possible without either introducing naming conflicts, or impossible circular dependencies.

File: Furball.Vixie.Helpers/DisposeQueue.cs (hunk 6 above)
Comment (Eeveelution) [reply Furball.Vixie.Helpers/DisposeQueue.cs:24]: tbh, now that i think bout it, its fine

**Review by Eeveelution — APPROVED** [review]
i suppose the lock/unlock thing being public can be left like that, unless you force yourself onto it it won't be a big issue
